mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -1341,6 +1358,10 @@ impl ObjectClient for S3CrtClient {
     }
 }
 
+pub trait OnTelemetry: std::fmt::Debug + Send + Sync {
+    fn on_telemetry(&self, request_metrics: &RequestMetrics, span: &Span);
```

Comment:
We will always have already entered the span before making this call, so is there any reason to also pass it explicitly?

Reply:
I kept it because it looks like some metadata is stored within the span

```
let op = span_telemetry.metadata().map(|m| m.name()).unwrap_or("unknown");
```

Not sure if it's useful though

Reply:
Ah right. I think we should probably just pass that `op` as an argument to the call; I think it's the only important one. The callback can always retrieve the span with [`Span::current`](https://docs.rs/tracing/0.1.40/tracing/struct.Span.html#method.current) if they need it.

Reply:
Cool I've updated the function signature to the operation as a &str. You're right, if callers really care then can grab it themselves.